src/atomEffect.ts

```diff
@@ -56,15 +56,13 @@ export function atomEffect(
       }
     },
     (get, set, ...args: Parameters<Setter>) => {
-      let result
       const ref = get(refAtom)
       ++ref.inProgress
       try {
-        result = set(...args)
+        return set(...args)
```

Comment:
I think returning here should work fine.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch
> Immediately before a control-flow statement (return, throw, break, continue) is executed in the try block or catch block.